### DISCUSSION TITLE
stft implementation

### DISCUSSION THREAD

**Discussion by dieter-wilhelm** [gjmvanboxtel/gsignal#11] [Ideas]
Hi

I think it would be more general to implement `stft` in the line of Matlab's `stft` (please check its [documentation](https://uk.mathworks.com/help/signal/ref/stft.html) ), which outputs complex amplitudes.  The power spectrum could be implemented optionally (or vice versa)..

What do you think?

   Dieter     

PS: I can't find  `stft` as reference in the [signal ](https://octave.sourceforge.io/signal/overview.html) package of octave.

**Comment by gjmvanboxtel** [comment]
Hi Dieter,

Yes, you are quite right. The Matlab version returns complex numbers. The Octave version returns the magnitude (https://octave.sourceforge.io/octave/function/stft.html). I followed to the Octave implementation for now, but perhaps in a future version this should change.

Best,
Geert

**Reply by dieter-wilhelm** [reply]
Hi Geert,

thank you for the octave `stft` link which I couldn't find.  It's alright if you model it on the Octave implementation.
 
Best wishes

  Dieter
